moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1478,32 +1484,34 @@ bool PlanningScene::processAttachedCollisionObjectMsg(const moveit_msgs::Attache
       return false;
     }
 
-    const robot_model::LinkModel* lm = getRobotModel()->getLinkModel(object.link_name);
-    if (lm)
+    const robot_model::LinkModel* link_model = getRobotModel()->getLinkModel(object.link_name);
+    if (link_model)
     {
-      std::vector<shapes::ShapeConstPtr> shapes;
-      EigenSTL::vector_Isometry3d poses;
+      collision_detection::World::Object object_to_attach(object.object.id);
```

Comment:
I don't see a benefit of introducing this World::Object as a container for shapes, poses, and subframes.
Keeping the old individual variables produces much less noise in this PR.

Reply:
I added that to express both the intention and the provenance/target of each of the operations throughout the next 100+ lines of code. I found that I had to scroll up and down quite a lot to understand all the assignments of shapes and poses.

With the individual variables, the first time they are used for their purpose is [150 lines further down](https://github.com/ros-planning/moveit/compare/master...ubi-agni:new-add-named_frames-to-collision-objects#diff-8aac12a54166575f8df07630635ac4ffR1647) :(

I would argue for leaving it in for readability. Processing the message isn't a frequent operation that needs to be optimized for performance.

Reply:
I cannot follow this argumentation: The old variables were introduced in line 1484+1485 and first filled in lines 1498+1499. Your new variable is introduced in line 1490 and filled in 1506+1507 - pretty much the same distance. Of course, the first call to `attachBody` is far away, but that didn't change as well. In the end, you just introduced another variable...